server/py/framework/db/sqldb/db.py

```diff
@@ -3191,6 +3206,79 @@ def _calculate_runs_counters(
             project_to_running_runs_count,
         )
 
+    def _calculate_alerts_activations_counters(
+        self,
+        session,
+    ) -> tuple[
+        dict[str, int],
+        dict[str, int],
+        dict[str, int],
+    ]:
+        projects_with_creation_time = self.list_projects(
+            session,
+            format_=mlrun.common.formatters.ProjectFormat.name_and_creation_time,
+        ).projects
+
+        project_to_endpoint_alerts_count = collections.defaultdict(int)
+        project_to_job_alerts_count = collections.defaultdict(int)
+        project_to_other_alerts_count = collections.defaultdict(int)
+
+        project_filter_conditions = [
+            and_(
+                AlertActivation.project == project,
+                AlertActivation.activation_time > created,
+            )
+            for project, created in projects_with_creation_time
+        ]
```

Comment:
This part (together with `.filter(or_(*project_filter_conditions))` can be moved to separate part and be used in both this funciton and list_alert_activations